xchange-examples/src/main/java/org/knowm/xchange/examples/blockchain/trade/BlockchainTradeDemo.java

```diff
@@ -0,0 +1,109 @@
+        System.out.println("===== TRADE SERVICE =====");
+        tradeServiceDemo();
+    }
+
+    @SneakyThrows
+    private static void tradeServiceDemo() throws IOException {
+        TradeService tradeService = BLOCKCHAIN_EXCHANGE.getTradeService();
+
+        System.out.println("===== placeLimitOrder =====");
+        LimitOrder limitOrder =
+                new LimitOrder.Builder(Order.OrderType.BID, CurrencyPair.ADA_USDT)
+                        .originalAmount(new BigDecimal("1.0"))
+                        .limitPrice(new BigDecimal("0.23"))
+                        .build();
+
+        String tradeLimitOrder = tradeService.placeLimitOrder(limitOrder);
+        System.out.println(OBJECT_MAPPER.writeValueAsString(tradeLimitOrder));
+
+        System.out.println("===== placeMarketOrder =====");
+        MarketOrder marketOrder =
+                new MarketOrder.Builder(Order.OrderType.BID, CurrencyPair.ADA_USDT)
+                        .originalAmount(new BigDecimal("1.0"))
+                        .cumulativeAmount(new BigDecimal("0.22"))
+                        .build();
+
+        String tradeMarketOrder = tradeService.placeMarketOrder(marketOrder);
+        System.out.println(OBJECT_MAPPER.writeValueAsString(tradeMarketOrder));
+
+        System.out.println("===== placeStopOrder =====");
+        StopOrder stopOrder =
+                new StopOrder.Builder(Order.OrderType.BID, CurrencyPair.ADA_USDT)
+                        .originalAmount(new BigDecimal("1.0"))
+                        .stopPrice(new BigDecimal("0.21"))
+                        .build();
+
+        String tradeStopOrder = tradeService.placeStopOrder(stopOrder);
+        System.out.println(OBJECT_MAPPER.writeValueAsString(tradeStopOrder));
+
+        Thread.sleep(5000);
+        System.out.println("===== getOpenOrders =====");
+
+        try {
+            final OpenOrdersParamCurrencyPair openOrdersParamsBtcUsd =
+                    (OpenOrdersParamCurrencyPair) tradeService.createOpenOrdersParams();
+            openOrdersParamsBtcUsd.setCurrencyPair(CurrencyPair.ADA_USDT);
+            OpenOrders openOrdersParams = tradeService.getOpenOrders(openOrdersParamsBtcUsd);
+            System.out.println(OBJECT_MAPPER.writeValueAsString(openOrdersParams));
+
+            OpenOrders openOrders = tradeService.getOpenOrders();
+            System.out.println(OBJECT_MAPPER.writeValueAsString(openOrders));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+        System.out.println("===== cancelOrder =====");
+        CancelOrderByCurrencyPair cancelOrderByCurrencyPair = () -> new CurrencyPair("ADA/USDT");
+        Boolean cancelAllOrderByCurrency = tradeService.cancelOrder(cancelOrderByCurrencyPair);
+        System.out.println("Canceling returned " + cancelAllOrderByCurrency);
+
+        System.out.println("===== getTradeHistory =====");
+
+        TradeHistoryParams params = tradeService.createTradeHistoryParams();
```

Comment:
make the service to return our BlockchainTradeHistoryParam, so you do not need all this casting